### ISSUE TITLE
[member-delimiter-style] allow delimiter to be specified for single lines

### ISSUE THREAD

**Issue by novemberborn** [bradzacher/eslint-plugin-typescript#92]
I have these type declarations:

```ts
export type ResolutionMap = Map<Config, {
  plugins: Map<string, Entry>
  presets: Map<string, Entry>
}>

const possibleNames: {
  fromFile: boolean
  name: string
}[] = []
```

I'd like to write them like:

```ts
export type ResolutionMap = Map<Config, {plugins: Map<string, Entry>; presets: Map<string, 
Entry>}>

const possibleNames: {fromFile: boolean; name: string}[] = []
```

However the `member-delimiter-style` configuration doesn't allow for single-line specific configuration.

My current options are `'{delimiter: 'none', requireLast: true, ignoreSingleLine: true}`.

I'd like to give this a try if folks think it's a good idea.

**Comment by mightyiam** [comment]
So, currently, if `delimiter: 'none'` and `ignoreSingleLine: false` then you just can't use the single line syntax at all?

**Comment by novemberborn** [comment]
Yup. And as I was going through my code, I noticed that's even more annoying when typing return values with say two object properties.

**Comment by mightyiam** [comment]
How about:

- remove `ignoreSingleLine`
- a new `singleLineDelimiter`
  - it can't be `'none'`, because that doesn't make sense
  - it does not inherit the value of `delimiter`, because `delimiter` can be `'none'`
  - sane default value
    - perhaps `'ignore'`?

**Comment by weirdpattern** [comment]
I like the approach suggested by @mightyiam 

But you might not even need the 'ignore' option at all. 
Just don't do any validation if singleLineDelimiter is not defined

**Comment by mightyiam** [comment]
```ts

  const breadcrumbs$ = pageSinks$
    .map(({ breadcrumbs }): Stream<Array<{ href: string; title: string }>> => breadcrumbs)
    .flatten()
```

That semicolon in `href: string;` is a hit.

Just another example of this issue.
